extensions/cl_ext_buffer_device_address.asciidoc

```diff
@@ -0,0 +1,289 @@
+   CL_MEM_DEVICE_ADDRESS_EXT. If the buffer was not created with the flag
+   or there are multiple devices in the context and the buffer address is
+   not the same for all of them, it returns CL_INVALID_MEM_OBJECT.
+
+| {CL_MEM_DEVICE_PTRS_EXT_anchor}
+include::{generated}/api/version-notes/CL_MEM_DEVICE_PTRS_EXT.asciidoc[]
+  | {cl_mem_device_address_pair_EXT_TYPE}
+      | Returns the device-address pairs for all devices in the context.
+      The per-device addresses might differ when the buffer was allocated
+      with the CL_MEM_DEVICE_PRIVATE_EXT enabled.
+|====
+
+
+=== Section 5.9.2 - Setting Kernel Arguments
+
+Add a new kernel argument setter for device pointers <<setting-kernel-arguments, Section 5.9.2>>:
+
+To set a device pointer as the argument value for a specific argument of a
+kernel, call the function
+
+include::{generated}/api/protos/clSetKernelArgDevicePointerEXT.txt[]
+include::{generated}/api/version-notes/clSetKernelArgDevicePointerEXT.asciidoc[]
+
+  * _kernel_ is a valid kernel object.
+  * _arg_index_ is the argument index.
+    Arguments to the kernel are referred by indices that go from 0 for the
+    leftmost argument to _n_ - 1, where _n_ is the total number of arguments
+    declared by a kernel.
+  * _arg_value_ is the device pointer that should be used as the argument value for
+    argument specified by _arg_index_.
+    The device pointer specified is the value used by all API calls that enqueue
+    _kernel_ ({clEnqueueNDRangeKernel} and {clEnqueueTask}) until the argument
+    value is changed by a call to {clSetKernelArgSVMPointer} for _kernel_.
+    The device pointer can only be used for arguments that are declared to be a
+    pointer to `global` memory allocated with clCreateBuffer() with the
+    CL_MEM_DEVICE_ADDRESS_EXT flag. The pointer value specified as the argument value
+    can be the pointer to the beginning of the buffer or be a pointer offset into
+    the buffer region. The device pointer value must be naturally aligned according to
+    the argument's type.
+
+// refError
+
+{clSetKernelArgDevicePointerEXT} returns {CL_SUCCESS} if the function was executed
+successfully. Otherwise, it returns one of the following errors:
+
+  * {CL_INVALID_KERNEL} if _kernel_ is not a valid kernel object.
+  * {CL_INVALID_OPERATION} if no devices in the context associated with _kernel_ support
+    the device pointer.
+  * {CL_INVALID_ARG_INDEX} if _arg_index_ is not a valid argument index.
+  * {CL_INVALID_ARG_VALUE} if _arg_value_ specified is not a valid value.
+  * {CL_OUT_OF_RESOURCES} if there is a failure to allocate resources required
+    by the OpenCL implementation on the device.
+  * {CL_OUT_OF_HOST_MEMORY} if there is a failure to allocate resources
+    required by the OpenCL implementation on the host.
+
+Add a new flag to clSetKernelExecInfo for setting indirect device pointer access info <<clSetKernelExecInfo, List of supported param_name stable>>:
```

Comment:
`clSetKernelExecInfo` needs to be adjusted a little more. Currently it states: `CL_INVALID_OPERATION if no devices in the context associated with kernel support SVM.` This needs to change to account for devices not supporting SVM to weaken the condition. E.g.

`CL_INVALID_OPERATION if CL_KERNEL_EXEC_INFO_SVM_PTRS or CL_KERNEL_EXEC_INFO_SVM_FINE_GRAIN_SYSTEM are specified and no device in the context associated with the kernel supports SVM`

`CL_INVALID_OPERATION if CL_KERNEL_EXEC_INFO_DEVICE_PTRS_EXT is specified and no device in the context associated with the kernel supports bda`